packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -15,6 +16,23 @@ import type {
   UserProfileCtx,
 } from '../types';
 
+const createDynamicParamParser =
+  ({ regex }: { regex: RegExp }) =>
+  <T extends Record<any, any>>({ urlWithParam, entity }: { urlWithParam: string; entity: T }) => {
+    const match = regex.exec(urlWithParam);
+
+    if (match) {
+      const key = match[1];
+      if (key in entity) {
+        const value = entity[key] as string;
+        return urlWithParam.replace(match[0], value);
+      }
+    }
+    return urlWithParam;
+  };
+
+const parse = createDynamicParamParser({ regex: /:(\w+)/ });
```

Comment:
❓ Do we have a need for multiple parsers right now? If not, I'd prefer a more explicit/simple approach with a more explicit name for the `parse` function.

Reply:
I do like having the `createDynamicParamParser` util here. I renamed it to `populateParamFromObject`

Reply:
Okay! Can we move it to its own file and write a couple of unit tests covering it? I see the tests for ClerkUIComponentsContext but lets include a few tests to better document this one :)